rotkehlchen/chain/ethereum/modules/thegraph/balances.py

```diff
@@ -0,0 +1,122 @@
+import logging
+from collections import defaultdict
+from typing import TYPE_CHECKING
+
+from rotkehlchen.accounting.structures.balance import Balance
+from rotkehlchen.accounting.structures.evm_event import EvmProduct
+from rotkehlchen.accounting.structures.types import HistoryEventSubType, HistoryEventType
+from rotkehlchen.chain.ethereum.interfaces.balances import ProtocolWithBalance
+from rotkehlchen.chain.ethereum.utils import asset_normalized_value
+from rotkehlchen.chain.evm.tokens import get_chunk_size_call_order
+from rotkehlchen.chain.evm.types import string_to_evm_address
+from rotkehlchen.constants.assets import A_GRT
+from rotkehlchen.db.dbhandler import DBHandler
+from rotkehlchen.fval import FVal
+from rotkehlchen.inquirer import Inquirer
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+
+from .constants import CPT_THEGRAPH
+
+if TYPE_CHECKING:
+    from rotkehlchen.chain.ethereum.interfaces.balances import BalancesType
+    from rotkehlchen.chain.evm.node_inquirer import EvmNodeInquirer
+    from rotkehlchen.types import CHAIN_IDS_WITH_BALANCE_PROTOCOLS
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+class ThegraphBalances(ProtocolWithBalance):
+    def __init__(
+            self,
+            database: DBHandler,
+            evm_inquirer: 'EvmNodeInquirer',
+            chain_id: 'CHAIN_IDS_WITH_BALANCE_PROTOCOLS',
+    ):
+        super().__init__(
+            database=database,
+            evm_inquirer=evm_inquirer,
+            chain_id=chain_id,
+            counterparty=CPT_THEGRAPH,
+        )
+        self.grt = A_GRT.resolve_to_evm_token()
+        self.staking_contract = string_to_evm_address('0xF55041E37E12cD407ad00CE2910B8269B01263b9')
+
+    def get_deposit_events(self) -> set[tuple[HistoryEventType, HistoryEventSubType]]:
+        return {(HistoryEventType.STAKING, HistoryEventSubType.DEPOSIT_ASSET)}
+
+    def query_balances(self) -> 'BalancesType':
+        balances: BalancesType = defaultdict(lambda: defaultdict(Balance))
+        grt_price = Inquirer().find_usd_price(A_GRT)
```

Comment:
Move this next to where it is used. at the moment you are always querying the price but it might be possible that you don't need it